### PR TITLE
drivers: adc: stm32: teardown internal channel after read

### PR DESCRIPTION
**Pull request by ycsin**
see https://github.com/zephyrproject-rtos/zephyr/pull/47691

### CODE REVIEW

**Review by aurel32 — DISMISSED** [review]
LGTM, thanks for the rebase.

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -582,6 +582,64 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 #endif /* LL_ADC_CHANNEL_VBAT */
 }
 
+static void adc_stm32_unset_common_path(const struct device *dev, uint32_t PathInternal)
+{
+	const struct adc_stm32_cfg *config = dev->config;
+	ADC_TypeDef *adc = (ADC_TypeDef *)config->base;
+	const uint32_t currentPath = LL_ADC_GetCommonPathInternalCh(__LL_ADC_COMMON_INSTANCE(adc));
+
+	ARG_UNUSED(adc); /* Avoid 'unused variable' warning for some families */
```
Comment (FRASTM) [inline drivers/adc/adc_stm32.c:591]: _adc_ is still used below

File: drivers/adc/adc_stm32.c [hunk 2]
```diff
@@ -521,7 +521,7 @@ static void adc_stm32_set_common_path(const struct device *dev, uint32_t PathInt
 	const struct adc_stm32_cfg *config =
 		(const struct adc_stm32_cfg *)dev->config;
 	ADC_TypeDef *adc = (ADC_TypeDef *)config->base;
-	(void) adc; /* Avoid 'unused variable' warning for some families */
+	ARG_UNUSED(adc); /* Avoid 'unused variable' warning for some families */
```
Comment (erwango) [inline drivers/adc/adc_stm32.c:524]: adc is used just below.

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by ycsin** [comment]
image

Looks like the `ARG_UNUSED` is indeed required by some series?

**Comment by erwango** [comment]
@ycsin can you revert you change as it appear ARG_UNUSED is indeed required.
I think the solution to fix this will be an overhaul of the driver to take better advantage of device tree.

**Comment by ycsin** [comment]
Added `ARG_UNUSED` back to the functions

**Review by aurel32 — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Review by str4t0m — APPROVED** [review]

**Pull request merged**